include/libdnf/rpm/repo.hpp

```diff
@@ -118,262 +109,349 @@ struct Repo {
     };
 
 
-    /**
-    * @brief Verify repo ID
-    *
-    * @param id repo ID to verify
-    * @return   index of the first invalid character in the repo ID (if present) or -1
-    */
-    static int verifyId(const std::string & id);
-
-    /**
-    * @brief Construct the Repo object
-    *
-    * @param id     repo ID to use
-    * @param conf   configuration to use
-    */
-    Repo(const std::string & id, std::unique_ptr<ConfigRepo> && conf, Repo::Type type = Repo::Type::AVAILABLE);
+    /// Verify repo ID
+    /// @param id repo ID to verify
+    /// @return   index of the first invalid character in the repo ID (if present) or -1
+    /// @replaces libdnf:repo/Repo.hpp:method:Repo.verifyId(const std::string & id)
+    static int verify_id(const std::string & id);
+
+    /// Construct the Repo object
+    /// @param id     repo ID to use
+    /// @param conf   configuration to use
+    /// @param base   reference to Base instance
+    /// @param type   type of repo
```

Comment:
Why type is called type? It sounds more like status. Also why type is part of the constructor?

Reply:
Please could you add TODO?

Reply:
Yes.